pkg/helm/helm.go

```diff
@@ -115,61 +104,23 @@ func (h *Hydrator) templateArgs(ctx context.Context, destDir string) ([]string,
 }
 
 func (h *Hydrator) appendValuesArgs(args []string) ([]string, error) {
-	switch h.ValuesFileApplyStrategy {
-
-	case "", ValuesFileApplyStrategyOverride:
-		for _, vs := range h.ValuesFilePaths {
-			if vs == "" {
-				continue
-			}
-			args = append(args, "--values", vs)
+	for _, vs := range h.ValuesFilePaths {
+		if vs == "" {
+			continue
 		}
+		args = append(args, "--values", vs)
+	}
 
-		if len(h.ValuesYAML) != 0 {
-			// inline values are passed in as a literal string, so we
-			// must write them out to a file
-			valuesPath, err := writeValuesPath([]byte(h.ValuesYAML))
-			if err != nil {
-				return nil, err
-			}
-			if valuesPath != "" {
-				args = append(args, "--values", valuesPath)
-			}
-		}
-
-	case ValuesFileApplyStrategyListConcatenate:
-		var valuesToMerge [][]byte
-		for _, vs := range h.ValuesFilePaths {
-			if vs == "" {
-				continue
-			}
-			val, err := readFile(vs)
-			if err != nil {
-				return nil, err
-			}
-			valuesToMerge = append(valuesToMerge, []byte(val))
-
-		}
-		if len(h.ValuesYAML) != 0 {
-			valuesToMerge = append(valuesToMerge, []byte(h.ValuesYAML))
-		}
-
-		merged, err := listConcatenate(valuesToMerge)
-		if err != nil {
-			return nil, fmt.Errorf("error merging values files: %w", err)
-		}
-		klog.Infof("using merged values: %s\n", string(merged))
-
-		valuesPath, err := writeValuesPath(merged)
+	if len(h.ValuesYAML) != 0 {
+		// inline yaml values are passed in as a literal string, so we
+		// must write them out to a file
+		valuesPath, err := writeValuesPath([]byte(h.ValuesYAML))
 		if err != nil {
 			return nil, err
 		}
 		if valuesPath != "" {
```

Comment:
will valuesPath ever be empty if err was nil?

Reply:
In the current implementation, if the inline values on the RSync are empty, then h.ValuesYAML will end up as an empty string, and `writeValuesPath` will return `"", nil`.

Reply:
aren't you protected from that with `if len(h.ValuesYAML) != 0 {` already?

Reply:
oh you're right, I can take out the duplicated check

Reply:
done